File: src/easy/RomanToInt.swift
import Foundation
import helpers

func romanToInt(roman: String) -> Int {
  var total = 0
  var index = roman.startIndex
  let romanNumerals = Utilities.romanNumeralsToArabic

  while index < roman.index(before: roman.endIndex) {
    let current = romanNumerals[roman[index], default: 0]
    let nextIndex = roman.index(after: index)
    let next = romanNumerals[roman[nextIndex], default: 0]
    if current < next {
      total -= current
    } else {
      total += current
    }
    index = nextIndex
  }
  return total + romanNumerals[roman[roman.index(before: roman.endIndex)], default: 0]
}

File: src/easy/roman-to-int.ts
const romanNumerals = new Map([
    ["I", 1],
    ["V", 5],
    ["X", 10],
    ["L", 50],
    ["C", 100],
    ["D", 500],
    ["M", 1_000],
]);
export const romanToInt = (roman: string): number => {
    const changeCase = roman.toUpperCase();
    return [...roman].reduce((a, c, i) => {
        const current = romanNumerals.get(c) ?? 0;
        const next = romanNumerals.get(changeCase[i + 1]) ?? 0;
        return current < next ? a - current : a + current;
    }, 0);
};
